cmd/snap/cmd_run.go

```diff
@@ -523,6 +523,14 @@ func (x *cmdRun) runCmdUnderStrace(origCmd, env []string) error {
 	go func() {
 		defer func() { filterDone <- true }()
 
+		// do not filter the initial strace output if SNAP_STRACE_SELF
+		// is set, this is useful when tracking down issues with snap
+		// helpers such as snap-confine, snap-exec ...
+		if osutil.GetenvBool("SNAP_STRACE_SELF", false) {
+			io.Copy(Stderr, stderr)
```

Comment:
This looks fine, but this is doing more than simply stracing self. This is stracing everything, and it's also disabling the filtering that we otherwise do. Assuming the _--raw_ option mentioned above, seems fine.